### DISCUSSION TITLE
Is it possible for a wind farm with the same type of wind turbines to have each turbine with a different TSR?

### DISCUSSION THREAD

**Discussion by Siantos** [NatLabRockies/floris#987] [Q&A]
i wanna to know，is it possible for a wind farm with the same type of wind turbines to have each turbine with a different TSR ? It makes different operation within turbines.

**Comment by Siantos** [comment]
Additionally, I found that the power of a wind turbine changes with the rotor diameter changed, but it remains the same when I change the value of TSR in the 'nrel_5mw.yaml' file. If I want to change the rotor speed of a particular wind turbine, thereby changing the TSR and consequently the turbine's output power, how should I write the code?

**Comment by misi9170** [comment]
Hi @Siantos,

Thanks for the question! While the TSR is an input on the turbine models, it currently only affects wakes in the GCH model (see [here](https://github.com/NREL/floris/blob/main/floris/core/wake_deflection/gauss.py#L293)), and effects on turbine power and thrust are not modeled. Generally speaking, FLORIS models the turbines themselves as actuator disks, which do not explicitly depend on TSR. 

That being said, we introduced the concept of "operation models" in FLORIS v4 ([documentation here](https://nrel.github.io/floris/operation_models_user.html)) partly to enable expanding above and beyond the basic actuator disk model (for example, see #832 and #924, which are still works in progress). If you're feeling adventurous, you could try implementing your own new operation model that took the TSR as an input, and varied the power and thrust outputs of the turbine accordingly. If you try that, we'd love to hear how you get on!

Cheers,
Misha
